### PR TITLE
Add benchmark templates and sample benches for IPA

### PR DESCRIPTION
**Pull request by mmagician**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: benches/benches.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// #![cfg(feature = "benches")]
```
Comment (Pratyush) [inline benches/benches.rs:1]:
```suggestion
```

File: Cargo.toml [hunk 2]
```diff
@@ -25,6 +25,8 @@ hashbrown = { version = "0.13", default-features = false, optional = true }
 digest = "0.10"
 derivative = { version = "2", features = [ "use_core" ] }
 rayon = { version = "1", optional = true }
+criterion = { version = "0.5", default-features = false, optional = true }
+rand_chacha = { version = "0.3.0", default-features = false, optional = true }
```
Comment (Pratyush) [inline Cargo.toml:29]: hm we don't want `criterion` as a normal dependency, as it has too many dependencies. `criterion` should be a dev-dependency.

Comment (Pratyush) [reply Cargo.toml:29]: Also, `rand_chacha` is already a dev-dependency.

File: src/bench_templates/mod.rs [hunk 3]
```diff
@@ -0,0 +1,241 @@
+use ark_crypto_primitives::sponge::{
```
Comment (Pratyush) [inline src/bench_templates/mod.rs:1]: We can make this a file in `benches`, or extract it into a separate crate (ala `ark-algebra-bench-templates`)

**Comment by mmagician** [comment]
Thanks @Pratyush, I've refactored the bench templates into its own crate as suggested. This required creating a workspace, so there's a bunch of files moved around, but ultimately the actual changes are few and much more elegant. Adding benches for a new scheme will also hopefully be simpler.

**Review by Pratyush — CHANGES_REQUESTED** [review]
Thanks for the change! The only thing that I think we should undo is the `pcs` renaming; it's fine to have the repo be called poly-commit and the folder within also called poly-commit.

**Review by Pratyush — APPROVED** [review]

**Pull request merged**